src/frontend/js/components/SaleTunnel/components/SaleTunnelStepResume/index.spec.tsx

```diff
@@ -26,4 +38,35 @@ describe('SaleTunnelStepResume', () => {
     fireEvent.click(button);
     expect(mockNext).toHaveBeenCalledTimes(1);
   });
+
+  it('shows a success message and a CTA to sign the order for a product with a contract definition', () => {
+    const mockNext = jest.fn();
+    const product = ProductFactory().one();
+    const order = OrderFactory().one();
+
+    render(
+      <IntlProvider locale="en">
+        <SaleTunnelContext.Provider
+          value={{
+            product,
+            order,
+            setOrder: noop,
+          }}
+        >
+          <SaleTunnelStepResume next={mockNext} />
+        </SaleTunnelContext.Provider>
+      </IntlProvider>,
+    );
+
+    const successLogo = screen.getByRole('img');
+    expect(successLogo).toBeInstanceOf(SVGElement);
+
+    screen.getByRole('heading', { level: 3, name: 'Congratulations!' });
+    screen.getByText(/In order to enroll to course runs you first need to sign the contract/);
+
+    // Click on the button trigger the next function
+    const button = screen.getByRole('link', { name: 'Go to signature' });
+    fireEvent.click(button);
```

Comment:
without awaiting the click event, mockNext may be called after test's end.
```suggestion
    const user = userEvent.setup()
    await user.click(button)
```